src/plugins/email/email.c

```diff
@@ -0,0 +1,94 @@
+/**
+ * @file
+ *
+ * @brief Source for email plugin
+ *
+ * @copyright BSD License (see doc/LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#include <kdberrors.h>
+#include <regex.h>
+#include <stdio.h>
+
+#include "email.h"
+
+static int validateEmail (const char * addr)
+{
+	if (!addr) return 0;
+	const char * regexString =
+		"^[a-zA-Z0-9!#$%&'*+\\/=?^_`{|}~-]+([a-zA-Z0-9!#$%&'*+\\/=?^_`{|}~.-][a-zA-Z0-9!#$%&'*+\\/=?^_`{|}~-])*"
+		"[a-zA-Z0-9!#$%&'*+\\/=?^_`{|}~-]?"
+		"@([a-zA-Z0-9]([a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)(\\.[a-zA-Z0-9]([a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$";
+	regex_t regex;
+	regmatch_t offsets;
+	int ret = regcomp (&regex, regexString, REG_NOSUB | REG_EXTENDED | REG_NEWLINE);
+	if (ret) return -1;
```

Comment:
also set proper error here.

Reply:
is it ok if the proper error is set outside like I changed i now, or does it need to be set in here?

Reply:
"here" can also be interpreted as "code path" but doing it directly in this line would have the advantage that you could use "regerror" to improve the error message easily. (But you could of course also somehow use negative special values to decide if regcomp or regexec failed).

Reply:
how about this?